tests/tests_pytorch/trainer/test_dataloaders.py

```diff
@@ -950,104 +950,87 @@ def test_dataloaders_load_only_once_no_sanity_check(tmpdir):
-    model.val_dataloader = Mock(wraps=model.val_dataloader)
-    model.test_dataloader = Mock(wraps=model.test_dataloader)
-
-    tracker.attach_mock(model.train_dataloader, "train_dataloader")
-    tracker.attach_mock(model.val_dataloader, "val_dataloader")
-    tracker.attach_mock(model.test_dataloader, "test_dataloader")
-
-    trainer.fit(model)
-    trainer.test(model)
-
-    # Verify the sequence
-    expected_sequence = [call.val_dataloader(), call.train_dataloader()]  # Sanity check first
-    if n == 1:
-        expected_sequence += [call.train_dataloader(), call.val_dataloader()] * 4
-    elif n == 2:
-        expected_sequence += [call.train_dataloader(), call.val_dataloader()] * 2
-    expected_sequence += [call.test_dataloader()]
-
-    assert tracker.mock_calls == expected_sequence
-
-    # Verify epoch of reloads
-    if n == 1:
-        assert train_reload_epochs == [0, 1, 2, 3, 4]
-        assert val_reload_epochs == [0, 1, 2, 3, 4]
-    elif n == 2:
-        assert train_reload_epochs == [0, 2, 4]
-        assert val_reload_epochs == [0, 2, 4]
-
-
 @pytest.mark.parametrize(
-    "n, train_reload_epochs_expect, val_reload_epochs_expect",
+    (
+        "num_sanity_val_steps, check_val_every_n_epoch, reload_dataloaders_every_n_epochs,"
+        " train_reload_epochs_expect,val_reload_epochs_expect,val_step_epochs_expect"
+    ),
     [
-        # Sanity check at epoch 0 creates a validation dataloader, but validation is
-        # checked (and in this case reloaded) every n epochs starting from epoch n-1
-        (3, [0, 2, 4, 6, 8], [0, 2, 5, 8]),
-        (5, [0, 2, 4, 6, 8], [0, 4, 9]),
+        # general case where sanity check reloads the dataloaders for validation on current_epoch=0
+        (0, 1, 1, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+        (1, 1, 1, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [1, 2, 3, 4, 5, 6, 7, 8, 9], [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+        # case where check_val_every_n_epoch < reload_dataloaders_every_n_epochs so expected val_reload_epoch
+        # and val_step_epoch will be different
+        (0, 1, 2, [0, 2, 4, 6, 8], [0, 2, 4, 6, 8], [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+        (1, 1, 2, [0, 2, 4, 6, 8], [2, 4, 6, 8], [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
+        (0, 3, 4, [0, 4, 8], [2, 8], [2, 5, 8]),
+        (1, 3, 4, [0, 4, 8], [2, 8], [2, 5, 8]),
+        # case where check_val_every_n_epoch > reload_dataloaders_every_n_epochs so expected val_reload_epoch
+        # and val_step_epoch will be same
+        (0, 2, 1, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [1, 3, 5, 7, 9], [1, 3, 5, 7, 9]),
+        (1, 2, 1, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [1, 3, 5, 7, 9], [1, 3, 5, 7, 9]),
+        (0, 3, 2, [0, 2, 4, 6, 8], [2, 5, 8], [2, 5, 8]),
+        (1, 3, 2, [0, 2, 4, 6, 8], [2, 5, 8], [2, 5, 8]),
+        (0, 5, 2, [0, 2, 4, 6, 8], [4, 9], [4, 9]),
+        (1, 5, 2, [0, 2, 4, 6, 8], [4, 9], [4, 9]),
```

Comment:
[this PR](https://github.com/Lightning-AI/lightning/issues/10948) introduced the link between `check_val_every_n_epoch` and `reload_dataloaders_every_n_epochs`. Here `reload_dataloaders_every_n_epochs` acts as a limit to check if this limit is crossed between consecutive validation. If `last validation epoch - current validation epoch >= reload_dataloaders_every_n_epochs`, then `check_val_every_n_epoch` takes the precedence.